apis/nucleus/src/components/listbox/interactions/listbox-keyboard-navigation.js

```diff
@@ -182,12 +193,20 @@ export function getListboxInlineKeyboardNavigation({
   const handleKeyDown = (event) => {
     const { keyCode, ctrlKey = false, shiftKey = false } = event.nativeEvent;
 
+    const container = event.currentTarget.closest('.listbox-container');
+
     switch (keyCode) {
-      // case KEYS.TAB: TODO: Focus confirm button using keyboard.focusSelection when we can access the useKeyboard hook.
+      case KEYS.TAB:
+        if (shiftKey) {
+          focusRow(container) || focusSearch(container);
+        } else {
+          focusSearch(container) || focusRow(container);
+        }
+        break;
       case KEYS.ENTER:
       case KEYS.SPACE:
-        if (!event.target.classList.contains('listbox-container')) {
-          return; // don't mess with keydown handlers within the listbox
+        if (!event.currentTarget.classList.contains('listbox-container')) {
```

Comment:
Actually, I removed this if-statement altogether since it causes scroll down when pressing Space inside of the confirm button (because the event is not stopped). For the other case, I use `currentTarget` since it is static. The latter refers to the element that the listener is bound to, which makes sense in that case, but not in the case you commented about.

Reply:
I would want to keep the check (event.target.classList.contains('listbox-container'))
for if to call keyboard.focus().
(I think it should only be done if the focus is on the listbox-container)

Reply:
Ok, yeah that makes sense actually. Also, I would expect this "Space"-key event to be stopped in the action toolbar and not here… I don't see why we would want "space" to continue propagating after a button has been pressed with space.